api/src/test/java/org/openmrs/api/db/hibernate/HibernateDiagnosisDAOTest.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * The contents of this file are subject to the OpenMRS Public License
+ * Version 1.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * http://license.openmrs.org
+ *
+ * Software distributed under the License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+ * License for the specific language governing rights and limitations
+ * under the License.
+ *
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+package org.openmrs.api.db.hibernate;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.openmrs.Diagnosis;
+import org.openmrs.api.db.DiagnosisDAO;
+import org.openmrs.test.BaseModuleContextSensitiveTest;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class HibernateDiagnosisDAOTest extends BaseModuleContextSensitiveTest {
+	
+	private static final String DIAGNOSIS_XML = "HibernateDiagnosisDAOTest-Dataset.xml";
+
+	@Rule
+	public ExpectedException thrown = ExpectedException.none();
+
+	@Autowired
+	DiagnosisDAO diagnosisDAO;
+
+	@Before
+	public void setUp() {
+		executeDataSet(DIAGNOSIS_XML);
+
+		updateSearchIndex();
+		
+	}
+
+	@Test
+	public void shouldDiagnosisByUuid() {
+		String uuid = "4e663d66-6b78-11e0-93c3-18a905e044dc";
+		Diagnosis diagnosis = diagnosisDAO.getDiagnosisByUuid(uuid);
+		assertEquals(diagnosis.getUuid(), uuid);
+	}
```

Comment:
You should be testing the service layer methods that delegate to these

Reply:
I actually feel that testing should be closer to the functionality we are testing. For instance, since this is testing data access methods, then the DAO test is more appropriate than the service layer test. If the service layer adds more business logic, then we can also have the service layer test.